lib/smart_answer_flows/marriage-abroad.rb

```diff
@@ -126,6 +126,13 @@ def define
           elsif calculator.partner_is_opposite_sex?
             if calculator.ceremony_country == 'hong-kong'
               outcome :outcome_opposite_sex_marriage_in_hong_kong
+            elsif calculator.ceremony_country == 'south-africa' &&
+                (
+                  calculator.resident_of_uk? ||
+                  calculator.resident_of_ceremony_country? ||
+                  calculator.resident_of_third_country?
+                )
```

Comment:
I agree with @leenagupte about removing this check. I think it makes the code more complex for no real benefit.